toolkit/tools/pkg/imagecustomizerlib/customizeos.go

```diff
@@ -30,20 +30,25 @@ func doOsCustomizations(ctx context.Context, rc *ResolvedConfig, imageConnection
 		return err
 	}
 
-	err = addRemoveAndUpdatePackages(ctx, rc.BuildDirAbs, rc.BaseConfigPath, rc.Config.OS, imageChroot, nil,
-		rc.Options.RpmsSources, rc.Options.UseBaseImageRpmRepos, distroHandler, rc.PackageSnapshotTime)
-	if err != nil {
-		return err
+	for _, configWithBase := range rc.ConfigChain {
+		err = addRemoveAndUpdatePackages(ctx, rc.BuildDirAbs, rc.BaseConfigPath, configWithBase.Config.OS,
+			imageChroot, nil, rc.Options.RpmsSources, rc.Options.UseBaseImageRpmRepos, distroHandler,
+			configWithBase.Config.OS.Packages.SnapshotTime)
```

Comment:
If you aren't going to use `rc.PackageSnapshotTime` here, then you probably need to refactor/delete the `validatePackageSnapshotTime` function. You also need to use `rc.Options.PackageSnapshotTime` somewhere.

This is likely the cause of your test failure.